pkg/queue/cluster_queue_impl.go

```diff
@@ -66,30 +65,21 @@ func (c *ClusterQueueImpl) DeleteFromQueue(q *Queue) {
 // pushIfNotPresent pushes the workload to ClusterQueue.
 // If the workload is already present, returns false. Otherwise returns true.
 func (c *ClusterQueueImpl) pushIfNotPresent(info *workload.Info) bool {
-	item := c.heap.items[workload.Key(info.Obj)]
-	if item != nil {
+	added, err := c.heap.AddIfNotPresent(info)
+	if err != nil {
 		return false
 	}
-	heap.Push(&c.heap, *info)
-	return true
+	return added
 }
 
 func (c *ClusterQueueImpl) PushOrUpdate(w *kueue.Workload) {
-	item := c.heap.items[workload.Key(w)]
-	info := *workload.NewInfo(w)
-	if item == nil {
-		heap.Push(&c.heap, info)
-		return
-	}
-	item.obj = info
-	heap.Fix(&c.heap, item.index)
+	info := workload.NewInfo(w)
+	_ = c.heap.AddOrUpdate(info)
 }
 
 func (c *ClusterQueueImpl) Delete(w *kueue.Workload) {
-	item := c.heap.items[workload.Key(w)]
-	if item != nil {
-		heap.Remove(&c.heap, item.index)
-	}
+	info := workload.NewInfo(w)
```

Comment:
Creating this object is not cheap. Can you remove using the key instead?

Reply:
good point!